### ISSUE TITLE
Experiment with epistemic uncertainty-aware training techniques

### ISSUE THREAD

**Issue by spencermyoung513** [porterjenkins/deep_uncertainty#16]
With the Double Poisson NLL, we would like to investigate training a Bayesian Neural Network. There are a variety of techniques we could try here, including:

- [ ] Variational inference
- [ ] Monte carlo dropout
